Sample.java

```diff
@@ -1,7 +1,71 @@
-// Time Complexity :
-// Space Complexity :
-// Did this code successfully run on Leetcode :
-// Any problem you faced while coding this :
+//Coin Change 2
+// Time Complexity :0(m*n)
+// Space Complexity :0(m*n)
+// Did this code successfully run on Leetcode :yes 
+// Any problem you faced while coding this :no
 
 
 // Your code here along with comments explaining your approach
+class Solution {
+    public int change(int amount, int[] coins) {
+        
+        int[][] dp= new int[coins.length+1][amount+1];
+        for(int i=0;i<=coins.length;i++)
+        {
+            dp[i][0]=1;
+        }
+        for(int j=1;j<=amount;j++)
+        {
+            dp[0][j]=0;
+        }
+        for(int i=1;i<=coins.length;i++)
+        {
+            for(int j=1;j<=amount;j++)
+            {
+                if(j<coins[i-1])
+                {
+                    dp[i][j]=dp[i-1][j];
+                }
+                else{
+                    dp[i][j]= dp[i-1][j]+dp[i][j-coins[i-1]];
+                }
+            }
+        }
+        return dp[coins.length][amount];
+    }
+}
+
+
+//paint house
+// Time Complexity :0(n)
+// Space Complexity :0(1)
+// Did this code successfully run on Leetcode :no,this question is locked
+// Any problem you faced while coding this : still not getting the correct output, don't lnow where I went wrong.
+class Solution {
+    public static int change( int[][] cost) {
+
+        if(cost.length==0)
+        {
+            return 0;
+        }
+        for(int i=1;i<cost.length;i++)
+        {
+            cost[i][0]=Math.min(cost[i-1][1],cost[i-1][2]);
```

Comment:
For the `i`th house, you have to add up the cost of painting `i`th house and the min cost of painting (i-1)th house.